libraries/transformer/src/main/java/androidx/media3/transformer/FrameworkMuxer.java

```diff
@@ -276,9 +283,64 @@ private static ImmutableList<String> getSupportedVideoSampleMimeTypes() {
     if (SDK_INT >= 24) {
       supportedMimeTypes.add(MimeTypes.VIDEO_H265);
     }
+    if (SDK_INT >= 33) {
+      supportedMimeTypes.add(MimeTypes.VIDEO_DOLBY_VISION);
+    }
     if (SDK_INT >= 34) {
       supportedMimeTypes.add(MimeTypes.VIDEO_AV1);
     }
     return supportedMimeTypes.build();
   }
+
+  // Get Dolby Vision profile
+  // Refer to https://professionalsupport.dolby.com/s/article/What-is-Dolby-Vision-Profile
+  @RequiresApi(33)
+  private static int getDvProfile(Format format) {
+    // Currently, only profile 8 is supported for encoding
+    // TODO: set profile ID based on format.
+    return MediaCodecInfo.CodecProfileLevel.DolbyVisionProfileDvheSt;
+  }
+
+  // Get Dolby Vision level
+  // Refer to https://professionalsupport.dolby.com/s/article/What-is-Dolby-Vision-Profile
+  @RequiresApi(33)
+  private static int getDvLevel(Format format) {
+    int level = -1;
+    int maxWidthHeight = Math.max(format.width, format.height);
+    float pps = format.width * format.height * format.frameRate;
+
+    if (maxWidthHeight <= 1280) {
+      if (pps <= 22118400) {
+        level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelHd24;  // Level 01
+      } else { // pps <= 27648000
+        level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelHd30;  // Level 02
+      }
+    } else if (maxWidthHeight <= 1920 && pps <= 49766400) {
+      level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelFhd24;  // Level 03
+    } else if (maxWidthHeight <= 2560 && pps <= 62208000) {
+      level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelFhd30;  // Level 04
```

Comment:
Do you know why the value of these constants is not same as mentioned level? For example DolbyVisionLevelFhd30 = 8

Reply:
Because the codec profile/level constants should be a bitmask, which is defined by Android AOSP. Dolby defined our profile/level to follow this rule. See https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/media/java/android/media/MediaCodecInfo.java;drc=6a4bef0f90e822e19866e53a98b85029bff04ea0;l=4301

I think Google defined it in this way due to efficiency. For example, to judge whether one profile is supported by current device, you just need to do one time comparison:
if (appointedLevel & (DolbyVisionLevelFhd24 | DolbyVisionLevelFhd24) {
    // Current device support this level
} else {
    // Current device doesn't support this level
}